ui/command_buffer_model.cpp

```diff
@@ -252,6 +253,16 @@ void CommandBufferModel::OnSelectionChanged(const QModelIndex &index)
     if (m_selected_node_index == selected_node_index)  // Selected same item
         return;
 
+    // Ensures that the selected index passed belongs to the filter model currently displayed in the
+    // main window.
+    QAbstractItemModel *model = const_cast<QAbstractItemModel *>(index.model());
```

Comment:
this seems to be risky. assuming model() returns const for a reason. is this really need?